core/integration/dind_test.go

```diff
@@ -0,0 +1,82 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/dagger/dagger/internal/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDIND(t *testing.T) {
+	t.Parallel()
+
+	var res struct {
+		Container struct {
+			From struct {
+				WithWorkDir struct {
+					WithExec struct {
+						WithNewFile struct {
+							WithExec struct {
+								Stdout string
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	err := testutil.Query(
+		`
+{
+  container {
+    from(address: "golang") {
+      withWorkdir(path: "/usr/src/app") {
+        withExec(args: ["sh", "-c", "go mod init test && go get dagger.io/dagger@main"]) {
```

Comment:
I think this will create a catch 22 if we make any backwards incompatible changes to the session interface since we will be testing the previous push to main against the dev engine here.

I think we can just load in the go sdk from `../../sdk/go` and setup a replace to use that instead

Reply:
> I think we can just load in the go sdk from `../../sdk/go` and setup a replace to use that instead

I guess that instead of doing this I can just use an `alpine` container and run a `curl` against the API. WDYT? Tests will be faster also since we don't need to pull the `golang` image also

Reply:
Yep that works too!

Reply:
Strange.. I tried changing the query to: 

```
{
  container {
    from(address: "alpine") {
      withExec(args: ["apk", "add", "curl"]) {
        withExec(args: ["sh", "-c", """
curl \
-u $DAGGER_SESSION_TOKEN: \
-H "content-type:application/json" \
-d '{"query":"{host{directory(path:\".\"){entries}}}"}' $DAGGER_SESSION_URL/query

        """], experimentalPrivilegedNesting: true) {
          stdout
        }
        }
    }
  }
}```

and I'm getting the same `context canceled` error as before :thinking:

Reply:
@marcosnils The new container you're using defaults the current work dir to `/`, so that `.` path is trying to load up the whole rootfs, including special filesystems like /proc which might result in the shim reading its own open file descriptors and other weird stuff. It seems to work as expected if you specify a normal directory like `/root` or similar.

The `context canceled` error is obviously not very helpful here though, I'm not sure why that's what we get. My best guess is something weird happening w/ proc or sys. I'll poke a bit and create a follow up issue if nothing else. But that can be a separate issue, shouldn't block this.